plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+If the data is not available, nodes should not attempt to use the commitment.
+If the data is only made available to a small portion of nodes, it should either be made available to all nodes by posting the data to L1 or the commitment should be rejected.
+Ensuring that nodes have a consistent view of the data is important to ensure that nodes derive the same chain and the fault proofs function correctly.
+
+Each DA Layer will have to implement logic on L1 to validate their commitments. This logic will emit a specific set of events that the op-node will read.
+These events then get adapted into a golang interface that the derivation pipeline uses to manage L2 reorgs & finality.
+
+### Data Availability Challenge Smart Contract Specification
+
+#### V1: Optimisic Bridge
+The DAC SC will emit the following event
+```solidity
+enum ChallengeStatus {
+    Uninitialized, // Never emitted
+    Active,        // Emitted when a challenge is initiated
+    Resolved,      // Emitted when a challenge is resolved. This means the data has been made available on L1.
+    Expired        // Never emitted
+}
+
+event ChallengeStatusChanged(bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber, ChallengeStatus status);
+```
+
+The flow is that after a commitment is posted to L1, it can be challenged within `challenge_window` blocks.
+If it is challenged an event with the `Active` status is emitted.
+If the challenge has not been resolved within `resolve_window` blocks, the challenge is presumed to succeed and the commitment is invalid.
+If the challenge is resolved within `resolve_window`, an event with the `Resolved` status is emitted and the commitment is valid.
+If a commitment is never challenged within `challenge_window` blocks, it is valid.
+
+This works essentially as an optimistic bridge. The commitment is assumed to be valid until someone claims that it is not.
+The validity is checked by a third party smart contract which understands the commitment format.
+
+
+#### V2: ZK Bridge
+
+DA Layers which have ZK bridges from their DA Chain to Ethereum can write a SC which emits the following event.
+
+```solidity
+    event Invalid(bytes challengedCommitment, uint256 challengedBlockNumber)
+    event Valid(bytes challengedCommitment, uint256 challengedBlockNumber)
+```
+
+The flow is that after a commitment is posted to L1, it can be challenged within `challenge_window` blocks.
+If the challenge succeeds, the `MarkedInvalid` event is emitted. The `MarkedInvalid` event should not be emitted unless the data is not available for all nodes.
+If the challenge fails, the `MarkedValid` event is emitted. If the `MarkedValid` event is emitted, the data MUST be available for all nodes.
+
+### Derivation Integration
+
+The op-node uses the following interface inside the derivation pipeline.
+```go
+type CommStatus int
+const (
+    OptimisticOK     = iota
```

Comment:
Nit: Could consider renaming to "AssumedValid," "OptimisticAssumedValid" or similar, which may be a bit more self-documenting than OptimisticOk and be more consistent with the other statuses.